_posts/2018-08-26-a-post-mortum-of-the-multiple-counting-bug-2018-03-28.md

```diff
@@ -0,0 +1,24 @@
+---
+layout: post
+title: A Post Mortem of The Multiple Counting Bug
+tags: [dev, core, crypto]
+author: dEBRUYNE
+---
+
+This blog sets out the multiple counting bug, of which two variants existed. The goal of this blog post is to provide a detailed explanation of aforementioned bug, how it was used to exploit services, merchants, and exchanges, and how it was handled by the Monero (dev) community.
+
+The multiple counting bug, of which two variants existed, was introduced in conjunction with the subaddress feature, which required a different (code) structure of the transaction public key. The first variant of the bug is basically that the code didn't impose a check to guard against duplicate public keys. Therefore, an attacker could create a transaction in which the transaction public key was included multiple times, thereby duplicating the particular transaction public key. As a result, the receiving wallet would report that it had received `x` times (where `x` is an integer that represents the number of identical transaction public keys) the amount it had actually received. All commands that report incoming transactions (e.g. `show_transfers` (CLI), `get_transfers` (RPC)) were affected. The balance, however, was not affected, i.e., the wallet would still report the balance properly. Alas, most exchanges utilize the `get_transfers` or `get_payments` wallet RPC command, which would, in case of duplicate transaction public keys, return an erroneous amount. 
+
+Now, unfortunately, this variant of the bug was fairly trivial to exploit, i.e., an attacker simply had to append `add_tx_pub_key_to_extra(tx, txkey_pub);` in `src/cryptonote_core/cryptonote_tx_utils.cpp` and his transaction public key would, after recompiling the code, be duplicated. Practically speaking this works as follows. The attackers starts with appending `src/cryptonote_core/cryptonote_tx_utils.cpp`, say, thrice^1, thereby creating four identical public transactions keys for his transactions. Subsequently, he sends a transaction of, say, 1 XMR to an exchange he seeks to exploit. The exchange will, most likely, credit the attacker with 4 XMR. The hacker, thereafter, withdraws this 4 XMR, thereby basically robbing the exchange of 3 XMR. If the exchange does not regularly check the balance against the sum of incoming and outgoing transfers or checks the hotwallet for any abnormalities, the hacker can basically repeat aforementioned process until the Monero hot wallet is emptied or, worst case scenario, all Monero funds are depleted. 
```

Comment:
"Now, unfortunately," -> "Unfortunately,"

Replace all instances of "his" with "their" and "he" with "they"